test/hotspot/jtreg/compiler/loopopts/parallel_iv/TestParallelIvInIntCountedLoop.java

```diff
@@ -0,0 +1,359 @@
+    @IR(failOn = {IRNode.COUNTED_LOOP})
+    private static long testIntCountedLoopWithLongIVZero(int stop) {
+        long a = 0;
+        for (int i = 0; i < stop; i++) {
+            a += 0;
+        }
+
+        return a;
+    }
+
+    @Test
+    @Arguments(values = {Argument.RANDOM_ONCE})
+    @IR(failOn = {IRNode.COUNTED_LOOP})
+    private static long testIntCountedLoopWithLongIVMax(int stop) {
+        long a = 0;
+        for (int i = 0; i < stop; i++) {
+            a += Long.MAX_VALUE;
+        }
+
+        return a;
+    }
+
+    @Test
+    @Arguments(values = {Argument.RANDOM_ONCE})
+    @IR(failOn = {IRNode.COUNTED_LOOP})
+    private static long testIntCountedLoopWithLongIVMaxMinusOne(int stop) {
+        long a = 0;
+        for (int i = 0; i < stop; i++) {
+            a += Long.MAX_VALUE - 1;
+        }
+
+        return a;
+    }
+
+    @Test
+    @Arguments(values = {Argument.RANDOM_ONCE})
+    @IR(failOn = {IRNode.COUNTED_LOOP})
+    private static long testIntCountedLoopWithLongIVMaxPlusOne(int stop) {
+        long a = 0;
+        for (int i = 0; i < stop; i++) {
+            a += Long.MAX_VALUE + 1;
+        }
+
+        return a;
+    }
+
+    @Test
+    @Arguments(values = {Argument.RANDOM_ONCE})
+    @IR(failOn = {IRNode.COUNTED_LOOP})
+    private static long testIntCountedLoopWithLongIVWithStrideTwo(int stop) {
+        long a = 0;
+        for (int i = 0; i < stop; i += 2) {
+            a += 2;
+        }
+
+        return a;
+    }
+
+    @Test
+    @Arguments(values = {Argument.RANDOM_ONCE})
+    @IR(failOn = {IRNode.COUNTED_LOOP})
+    private static long testIntCountedLoopWithLongIVWithStrideMinusOne(int stop) {
+        long a = 0;
+        for (int i = stop; i > 0; i += -1) {
+            a += 1;
+        }
+
+        return a;
+    }
+
+    @Test
+    @Arguments(values = {Argument.RANDOM_ONCE})
+    @IR(failOn = {IRNode.COUNTED_LOOP})
+    private static long testIntCountedLoopWithLongIVWithRandomStrides(int stop) {
+        long a = 0;
+        for (int i = 0; i < stop; i += stride) {
+            a += (long) stride2;
+        }
+
+        return a;
+    }
+
+    @Test
+    @Arguments(values = {Argument.RANDOM_ONCE, Argument.RANDOM_ONCE, Argument.RANDOM_ONCE})
+    @IR(failOn = {IRNode.COUNTED_LOOP})
+    private static long testIntCountedLoopWithLongIVWithRandomStridesAndInits(long init, int init2, int stop) {
+        long a = init;
+        for (int i = init2; i < stop; i += stride) {
+            a += stride2;
+        }
+
+        return a;
+    }
+
+    private static void testCorrectness() {
+        Random rng = new Random();
+
+        // Since we can't easily determined expected values if loop varibles overflow, we make sure i is less than (MAX_VALUE - stride).
+        int[] iterations = {0, 1, 2, 42, 100, rng.nextInt(0, Integer.MAX_VALUE - stride)};
+
+        for (int i : iterations) {
+            Asserts.assertEQ(i, testIntCountedLoopWithIntIV(i));
```

Comment:
Code in this loop is not guaranteed to be even C2 compiled because IR verification will be executed in a separate VM. IR framework tests that also want to verify the output, should be written like this:

https://github.com/openjdk/jdk/blob/9bd478593cc92a716151d1373f3426f1d92143bb/test/hotspot/jtreg/testlibrary_tests/ir_framework/examples/CustomRunTestExample.java#L84-L97

Reply:
Updated to use custom run methods instead. Thanks for the info!